### PR TITLE
Create org from register form and register from onboarding

### LINKED ISSUE #3742: Create org from register form and register from onboarding
Currently, users are asked to contact us for creating an organization on demo. But actually they can do that themselves with the onboarding flow. We can just link there.
bildschirmfoto 2019-02-06 um 17 27 21

Similarly, the onboarding flow should have a way to register/login instead of creating a new org. 

Relates to #3683 (somewhat).

### PR DESCRIPTION
**Pull request by daniel-wer**
Show a link to the onboarding flow when registering and `allowOrganizationCreation` is set.
Also allow to join an existing organization in the first onboarding step which will redirect the user to the register page with the organization pre-selected. This makes sense, because the `Get Started` button on instances with `allowOrganizationCreation` set is very eye-catching, so users may click it, although they actually want to simply register.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Set `allowOrganizationCreation` to true in the `application.conf`.
  - Visit the registration page, a link to the onboarding flow should be displayed.
  - In the first step of the onboarding flow:
    - Select an existing organization and click "Join" - You should be redirected to the registration page with the org prefilled.
    - Insert a new organization and click "Create" - The onboarding flow should continue as usual.

### Issues:
- fixes #3742 

------
- [x] Ready for review

### CODE REVIEW

File: app/assets/stylesheets/goldenlayout_overwrites.less [hunk 1]
```diff
@@ -64,9 +64,9 @@
       color: rgba(0, 0, 0, 0.65);
       font-size: 14px;
       font-weight: 400;
-      font-family: "Monospaced Number", "Chinese Quote", -apple-system, BlinkMacSystemFont,
-        "Segoe UI", Roboto, "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei", "Helvetica Neue",
-        Helvetica, Arial, sans-serif;
+      font-family: "Monospaced Number", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
```
Comment (daniel-wer) [inline app/assets/stylesheets/goldenlayout_overwrites.less]: This font caused the weird rendering of the left/right quotes. Unfortunately antd re-sets this font for some elements. Maybe we need to manually overwrite it for those cases.

File: app/assets/javascripts/admin/onboarding.js [hunk 2]
```diff
@@ -121,12 +135,21 @@ export class InviteUsersPopover extends React.Component<{
 }
 
 const OrganizationForm = Form.create()(({ form, onComplete }) => {
+  const [organizations, setOrganizations] = useState([]);
+  const fetchOrganizations = async () =>
+    setOrganizations((await getOrganizations()).map(org => org.name));
+  useEffect(() => {
```
Comment (philippotto) [inline app/assets/javascripts/admin/onboarding.js]: `useEffect(fetchOrganizations, [])` should work, too, right?

File: app/assets/javascripts/admin/onboarding.js [hunk 3]
```diff
@@ -121,12 +135,21 @@ export class InviteUsersPopover extends React.Component<{
...
+
+  const joinExistingOrganization = organizations.includes(form.getFieldValue("organizationName"));
+
   const hasErrors = fieldsError => Object.keys(fieldsError).some(field => fieldsError[field]);
   const handleSubmit = e => {
     e.preventDefault();
     form.validateFields((err, values) => {
       if (!err) {
-        onComplete(values.organizationName);
+        onComplete(values.organizationName, joinExistingOrganization);
```
Comment (philippotto) [inline app/assets/javascripts/admin/onboarding.js]: I think the second parameter is not used? Should be in Line 241, right?

File: app/assets/javascripts/admin/onboarding.js [hunk 4]
```diff
@@ -147,27 +170,36 @@ const OrganizationForm = Form.create()(({ form, onComplete }) => {
             {getFieldDecorator("organizationName", {
               rules: [{ required: true, message: "Please enter an organization name!" }],
             })(
-              <Input
-                size="large"
+              <AutoComplete
+                dataSource={organizations}
```
Comment (philippotto) [inline app/assets/javascripts/admin/onboarding.js]: Is there something, such as `size="large"` for `AutoComplete`? The old input box had the same height and font size as the button next to it. Now, it looks a bit weird :) See https://demo.webknossos.org/onboarding for comparison.

File: app/assets/javascripts/admin/onboarding.js (hunk 2 above)
Comment (daniel-wer) [reply app/assets/javascripts/admin/onboarding.js]: Unfortunately not ^.~

> There is one last catch. In the code, we are using async/await to fetch data from a third-party API. According to the documentation every function annotated with async returns an implicit promise: “The async function declaration defines an asynchronous function, which returns an AsyncFunction object. An asynchronous function is a function which operates asynchronously via the event loop, using an implicit Promise to return its result. “. However, an effect hook should return nothing or a clean up function. That’s why you may see the following warning in your developer console log: 07:41:22.910 index.js:1452 Warning: useEffect function must return a cleanup function or nothing. Promises and useEffect(async () => …) are not supported, but you can call an async function inside an effect.. That’s why using async directly in the useEffect function isn’t allowed. Let’s implement a workaround for it, by using the async function inside the effect.

from https://www.robinwieruch.de/react-hooks-fetch-data/

**Comment by daniel-wer** [comment]
> now, he is taken outside of the onboarding flow (the steps header is missing and he will never be led to "upload dataset" or "what's next")

Yes this is by design, as I thought when joining an existing organization, the user doesn't want to upload a dataset or learn about the admin functionality (in step 4). I would say the onboarding flow was designed with an admin in mind, but not for joining users. What do you think?

File: app/assets/javascripts/admin/onboarding.js (hunk 2 above)
Comment (philippotto) [reply app/assets/javascripts/admin/onboarding.js]: Argh, ok, this will still bite me a few times probably.

**Comment by philippotto** [comment]
> > now, he is taken outside of the onboarding flow (the steps header is missing and he will never be led to "upload dataset" or "what's next")
> 
> Yes this is by design, as I thought when joining an existing organization, the user doesn't want to upload a dataset or learn about the admin functionality (in step 4). I would say the onboarding flow was designed with an admin in mind, but not for joining users. What do you think?

Ah ok, I understand. Yes, you are right. The onboarding flow is more of an admin-thing. It could be adapted to be more generic, but that's probably not necessary for now.

**Comment by daniel-wer** [comment]
> * also, the registration form where he is led to says...
You were invited to join the organization “Connectomics_Department”!
 In case you do not know this organization, contact [hello@scalableminds.com](mailto:hello@scalableminds.com) to get more information about how to get to use webKnossos.
...which is weird.

I agree, would you prefer to adapt the copy so it fits both use cases (invitation and pre-selection of the organization) or should I hide the text, if the user pre-selected the organization or show a different copy in both cases? ^^

**Comment by philippotto** [comment]
> I agree, would you prefer to adapt the copy so it fits both use cases (invitation and pre-selection of the organization) or should I hide the text, if the user pre-selected the organization or show a different copy in both cases? ^^

I think, one copy which fits all cases would be ideal :)

**Comment by daniel-wer** [comment]
This proves a little difficult as it's hard to get the flow right. Ideally, the user should be able to choose another organization if the specified one is not the correct one, but that is currently not possible because the dropdown is hidden.

I would propose to always show the org dropdown and simply prefill it with the supplied organizationName if there is any, but I can also see the appeal of hiding the dropdown for "invited" users.

Maybe it's easiest to discuss this in person :)

**Comment by philippotto** [comment]
> Maybe it's easiest to discuss this in person :)

Yes, let's do that :)

**Comment by daniel-wer** [comment]
I adapted the copy according to our discussion :)

**Pull request merged**